### DISCUSSION TITLE
python message federate example

### DISCUSSION THREAD

**Discussion by nick-gorman** [GMLC-TDC/HELICS#1599] [Q&A] [answered]
Hi, 

I was wondering if anyone could point me in the direction of an example for creating message federates in python, similar to this example for value federates https://docs.helics.org/en/latest/introduction/python.html#python-example . Or even in another language would be fine, as long as the APIs are similar.

Cheers,
Nick

**Comment by trevorhardy** [comment, marked as answer]
We're in the process of creating a fleet of Python-based examples that will be linked to the User Guide (also being re-worked). If you look in the HELICS-examples repo,[the filter example ](https://github.com/GMLC-TDC/HELICS-Examples/tree/master/user_guide_examples/fundamental/fundamental_message_comm/filters) sends messages between [Charger.py](https://github.com/GMLC-TDC/HELICS-Examples/blob/bd82105cc23b31c91d60adecc781b0b00889ac90/user_guide_examples/Example41/Charger.py#L277) and [Controller.py](https://github.com/GMLC-TDC/HELICS-Examples/blob/bd82105cc23b31c91d60adecc781b0b00889ac90/user_guide_examples/Example41/Controller.py#L115). See also the same code to see how you can set up a [Python federate to receive messages](https://github.com/GMLC-TDC/HELICS-Examples/blob/bd82105cc23b31c91d60adecc781b0b00889ac90/user_guide_examples/Example41/Controller.py#L95).

(If you want to play along at home, we are adding the other examples in the `user_guide_examples` branch over the coming weeks and they'll eventually be moved into `master` and linked in the User Guide.)

**Reply by nick-gorman** [reply]
Awesome, thanks for the info!
